core/src/components/tab-bar/tab-bar.tsx

```diff
@@ -59,43 +60,32 @@ export class TabBar implements ComponentInterface {
   }
 
   connectedCallback() {
-    if (typeof (window as any) !== 'undefined') {
-      this.keyboardWillShowHandler = () => {
-        if (this.el.getAttribute('slot') !== 'top') {
-          this.keyboardVisible = true;
-        }
-      };
+    this.keyboardCtrl = createKeyboardController((keyboardOpen) => {
+      this.keyboardVisible = keyboardOpen; // trigger re-render by updating state
+    });
 
-      this.keyboardWillHideHandler = () => {
-        setTimeout(() => (this.keyboardVisible = false), 50);
-      };
-
-      window.addEventListener('keyboardWillShow', this.keyboardWillShowHandler!);
-      window.addEventListener('keyboardWillHide', this.keyboardWillHideHandler!);
-    }
+    this.keyboardCtrl.init();
```

Comment:
What's the reasoning for having an explicit init call rather than having the controller auto-init?

Reply:
Hm, I suppose the separate init isn't needed if the controller is created at the same time 🤔 I'll make it auto-init.